bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -220,20 +226,45 @@ protected static ScheduledExecutorService createDefaultRetryExecutor() {
   private final EventLoopGroup elg;
   private final ScheduledExecutorService scheduledRetries;
   private final List<Closeable> clientCloseHandlers = new ArrayList<>();
+  private final Future<ClientInterceptor> credentialInterceptorFuture;
 
   public BigtableSession(BigtableOptions options, ExecutorService batchPool) {
-    this(options, batchPool, createDefaultEventLoopGroup(), createDefaultRetryExecutor());
+    this(options, batchPool, null, null);
   }
 
-  public BigtableSession(BigtableOptions options, ExecutorService batchPool, EventLoopGroup elg,
-      ScheduledExecutorService scheduledRetries) {
-    this.elg = elg;
+  public BigtableSession(final BigtableOptions options, final ExecutorService batchPool,
+      EventLoopGroup elg, ScheduledExecutorService scheduledRetries) {
+    credentialInterceptorFuture = batchPool.submit(new Callable<ClientInterceptor>() {
+      @Override
+      public ClientInterceptor call() throws Exception {
+        final Credentials credentials = CredentialFactory.getCredentials(options
+            .getCredentialOptions());
+        if (credentials != null) {
+          ClientAuthInterceptor interceptor = new ClientAuthInterceptor(credentials, batchPool);
+          batchPool.execute(new Runnable() {
+            @Override
+            public void run() {
+              try {
+                credentials.refresh();
+              } catch (IOException e) {
+                LOG.error("Could not initialize the credentials", e);
+              }
+            }
+          });
+          return interceptor;
+        } else {
+          return null;
```

Comment:
The only time they can be null is for unit testing.  See CredentialFactory.nullCredential() for the warning.